leaphymicropython/utils/i2c_helper.py

```diff
@@ -1,22 +1,37 @@
 """This module provides helper functions for i2c."""  # Single-line docstring
 
 import struct
+from machine import Pin, I2C
+from leaphymicropython.utils.i2c_address_finder import is_device_address_visible
 
 
-def i2c_is_it_used_and_alive(func):
-    """check if i2c connection is used and if so, if it is alive before using it"""
+def handle_i2c_errors(func):
+    """
+    Decorator to handle I2C errors.
+
+    This decorator wraps a function that interacts with I2C devices.
+    It checks if the I2C connection is alive and attempts to reinitialize
+    the connection if necessary. It also handles specific OSError exceptions
+    that may occur during I2C communication.
 
+    Args:
+        func: The function to be wrapped.
+
+    Returns:
+        The wrapped function.
+    """
     def wrapper(*args, **kwargs):
         instance = args[0]
-        # check if i2c is used
-        i2c_used = hasattr(instance, "i2c")
-        if not i2c_used:
+        # check if the class instance is a subclass of I2CSensorOrActuator
+        if not isinstance(instance, I2CSensorOrActuator):
             result = func(*args, **kwargs)
```

Comment:
```suggestion
            return func(*args, **kwargs)
```

De else: kan dan weg, scheel 1 "level" en maakt het wat beter leesbaar.